icu4c/source/common/localebuilder.cpp

```diff
@@ -218,6 +219,7 @@ _clearUAttributesAndKeyType(Locale& locale, UErrorCode& errorCode)
 void
 _setUnicodeExtensions(Locale& locale, const CharString& value, UErrorCode& errorCode)
 {
+    if (U_FAILURE(errorCode)) { return; }
```

Comment:
optional: an example where _everything_ takes a UErrorCode

This is a nice example of how, when the stars align, this can work about as automatically as exceptions, when we can't actually use exceptions.